### ISSUE TITLE
Broken nullability contract, unable to init BugSnag

### ISSUE THREAD

**Issue by jbruchanov** [bugsnag/bugsnag-android#1073]
**This is a bug reproducible only in tests running via roboletric**.
So let's ignore for now, that it shouldn't be init in tests it was discovered accidentally... 
I still believe though there is a potentially issue in the code as it has broken nullability contract.

### Describe the bug
Unable to init BugSnag

### Steps to reproduce
Calling just `Bugsnag.start(this/*application*/)` in `App` instance

### Environment
* Bugsnag version: 5.5.0 (defined as 5.+)
* Running a test in roboletric 4.3.1

### Workaround
* Revert to 5.4.0 helps

---
The issue is here 

https://github.com/bugsnag/bugsnag-android/blob/next/bugsnag-android-core/src/main/java/com/bugsnag/android/Client.java#L159-L167

`DeviceIdStore.loadDeviceId(): String?` returns in my case `null` because of
```
java.io.IOException: The process cannot access the file because another process has locked a portion of the file
	at java.io.FileOutputStream.writeBytes(Native Method)
	at java.io.FileOutputStream.write(FileOutputStream.java:326)
	at sun.nio.cs.StreamEncoder.writeBytes(StreamEncoder.java:221)
	at sun.nio.cs.StreamEncoder.implClose(StreamEncoder.java:316)
	at sun.nio.cs.StreamEncoder.close(StreamEncoder.java:149)
	at java.io.OutputStreamWriter.close(OutputStreamWriter.java:233)
	at java.io.BufferedWriter.close(BufferedWriter.java:266)
	at kotlin.io.CloseableKt.closeFinally(Closeable.kt:57)
	at com.bugsnag.android.SynchronizedStreamableStore.persist(SynchronizedStreamableStore.kt:24)
	at com.bugsnag.android.DeviceIdStore.persistNewDeviceIdWithLock(DeviceIdStore.kt:123)
	at com.bugsnag.android.DeviceIdStore.persistNewDeviceUuid(DeviceIdStore.kt:97)
	at com.bugsnag.android.DeviceIdStore.loadDeviceId$bugsnag_android_core_release(DeviceIdStore.kt:65)
	at com.bugsnag.android.DeviceIdStore.loadDeviceId(DeviceIdStore.kt:48)
	at com.bugsnag.android.Client.<init>(Client.java:159)
	at com.bugsnag.android.Bugsnag.start(Bugsnag.java:65)
	at com.bugsnag.android.Bugsnag.start(Bugsnag.java:41)
	....
```
As you can see for some reason (probably robolectric related), it can't persist the `deviceId` and in this case returns `null`, which is unexpected later.

and crashing later on because of 
`DeviceDataCollector.ctor(...)` expects `installId: String`
```
java.lang.IllegalArgumentException: Parameter specified as non-null is null: method com.bugsnag.android.DeviceDataCollector.<init>, parameter installId

	at com.bugsnag.android.DeviceDataCollector.<init>(DeviceDataCollector.kt)
	at com.bugsnag.android.Client.<init>(Client.java:167)
	at com.bugsnag.android.Bugsnag.start(Bugsnag.java:65)
	at com.bugsnag.android.Bugsnag.start(Bugsnag.java:41)
	at ....App.initBugsnag(App.kt:83)
	at ....App.onCreate(App.kt:58)
	at org.robolectric.android.internal.AndroidTestEnvironment.lambda$installAndCreateApplication$0(AndroidTestEnvironment.java:276)
	at org.robolectric.util.PerfStatsCollector.measure(PerfStatsCollector.java:75)
	at org.robolectric.android.internal.AndroidTestEnvironment.installAndCreateApplication(AndroidTestEnvironment.java:276)
	at org.robolectric.android.internal.AndroidTestEnvironment.setUpApplicationState(AndroidTestEnvironment.java:169)
	at org.robolectric.RobolectricTestRunner.beforeTest(RobolectricTestRunner.java:301)
	at org.robolectric.internal.SandboxTestRunner$2.lambda$evaluate$0(SandboxTestRunner.java:243)
	at org.robolectric.internal.bytecode.Sandbox.lambda$runOnMainThread$0(Sandbox.java:89)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

So even if it's most likely just a crash because of unexpected execution environment, that broken nullability contract is a potentially bug if `deviceId` can't be persisted for whatever reason and should probably crash with better explanation or should crash silently with error logging.

**Comment by mattdyoung** [comment]
Hi @jbruchanov 

Thanks for the report, we'll take a look at fixing this. If you did have any other information about the conditions under which the IOException was thrown that would also be useful.

**Comment by phillipsam** [comment]
A fix for this has now been released in [v5.5.1](https://github.com/bugsnag/bugsnag-android/releases/tag/v5.5.1).
